### DISCUSSION TITLE
How to implement link shortening as a blog widget?

### DISCUSSION THREAD

**Discussion by herovn9090** [YOURLS/YOURLS#3119] [Q&A] [answered]
I have successfully installed yourl on the blog, but now only I can access the shorten link function at https://myblog.com/admin with the user and password pre-set in the file. configure config.php.

Now I want to expand to provide this utility on the blog as follows:

1) Create a Tool page, for example: https://myblog.com/shortenlink, this page is a static page on my main blog https://myblog.com.

2) Allow all visitors to https://myblog.com/shortenlink without username and password.

3) Design and supplement information content to guide visitors to use this service.

Please guide me.

TKs.

**Comment by dgw** [comment, marked as answer]
You are asking for a [public interface](https://github.com/YOURLS/YOURLS/wiki/Public-Shortening#public-interface).
